### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by jaymell**
Add documentation for `cluster_async` module to code and README;
additional small updates elsewhere for clarity.

### CODE REVIEW

**Review by shachlanAmazon — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ fn fetch_an_integer() -> redis::RedisResult<isize> {
 
 ## Async support
 
-To enable asynchronous clients a feature for the underlying feature need to be activated.
+To enable asynchronous clients, you must enable the relevant feature in your Cargo.toml.
```
[the remark this answers is not in the document]
Comment (jaymell) [reply README.md:53]: I like your suggest to change from passive to active voice, but not sure what to do different about mentioning the feature name. We mention tokio/async-std features immediately after this sentence, right?

[the remark this answers is not in the document]
Comment (nihohit) [reply README.md:53]: Asynchronous client support can be enabled by adding the `tokio/async-std-comp` feature, depending on your async runtime. See example:

[the remark this answers is not in the document]
Comment (nihohit) [reply README.md:53]: Asynchronous client support can be enabled by adding the tokio/async-std-comp feature, depending on your async runtime. See example:

[the remark this answers is not in the document]
Comment (shachlanAmazon) [reply README.md:53]: Asynchronous client support can be enabled by adding the tokio/async-std-comp feature, depending on your async runtime. See example:

[the remark this answers is not in the document]
Comment (jaymell) [reply README.md]: Thanks -- updated.

**Pull request merged**